incremental-font-transfer/src/glyph_keyed.rs

```diff
@@ -540,36 +489,144 @@ enum OffsetType {
     Long,
 }
 
+trait OffsetTypeInfo {
+    fn width() -> usize;
+    fn divisor() -> usize;
+    fn bias() -> u32;
+}
+
+struct CffOneInfo;
+
+impl OffsetTypeInfo for CffOneInfo {
+    fn width() -> usize {
+        1
+    }
+
+    fn divisor() -> usize {
+        1
+    }
+
+    fn bias() -> u32 {
+        1
+    }
+}
+
+struct CffTwoInfo;
+
+impl OffsetTypeInfo for CffTwoInfo {
+    fn width() -> usize {
+        2
+    }
+
+    fn divisor() -> usize {
+        1
+    }
+
+    fn bias() -> u32 {
+        1
+    }
+}
+
+struct CffThreeInfo;
+
+impl OffsetTypeInfo for CffThreeInfo {
+    fn width() -> usize {
+        3
+    }
+
+    fn divisor() -> usize {
+        1
+    }
+
+    fn bias() -> u32 {
+        1
+    }
+}
+
+struct CffFourInfo;
+
+impl OffsetTypeInfo for CffFourInfo {
+    fn width() -> usize {
+        4
+    }
+
+    fn divisor() -> usize {
+        1
+    }
+
+    fn bias() -> u32 {
+        1
+    }
+}
+
+struct ShortDivByTwoInfo;
+impl OffsetTypeInfo for ShortDivByTwoInfo {
+    fn width() -> usize {
+        2
+    }
+
+    fn divisor() -> usize {
+        2
+    }
+
+    fn bias() -> u32 {
+        0
+    }
+}
+
+struct LongInfo;
+impl OffsetTypeInfo for LongInfo {
+    fn width() -> usize {
+        4
+    }
+
+    fn divisor() -> usize {
+        1
+    }
+
+    fn bias() -> u32 {
+        0
+    }
+}
+
 impl OffsetType {
-    fn max_representable_size(self) -> u64 {
+    fn max_representable_size(self) -> usize {
         match self {
             Self::ShortDivByTwo => ((1 << 16) - 1) * 2,
-            _ => (1 << (self.offset_width() * 8)) - 1 - self.offset_bias() as u64,
+            _ => (1 << (self.offset_width() * 8)) - 1 - self.offset_bias() as usize,
         }
     }
 
     fn offset_width(&self) -> usize {
         match self {
-            Self::CffOne => 1,
-            Self::CffTwo => 2,
-            Self::CffThree => 3,
-            Self::CffFour => 4,
-            Self::ShortDivByTwo => 2,
-            Self::Long => 4,
+            Self::CffOne => CffOneInfo::width(),
```

Comment:
could we replace all this with an  `info()` method that returns the relevant `OffsetTypeInfo`?

Reply:
info() would need to return a "impl OffsetTypeInfo" which won't work since that can't be used when different concrete types may be returned. I tried to figure out a way to get this to work but wasn't able to. Any ideas on how to do it?

Reply:
Made this a little bit better by adding a concrete offset info instance into each enum. Still need the matches though.